lib/models/blueprint.js

```diff
@@ -850,7 +850,9 @@ Blueprint.prototype._locals = function(options) {
   var sanitizedModuleName = moduleName.replace(/\//g, '-');
 
   return new Promise(function(resolve) {
-    resolve(this.locals(options));
+    return resolve(this._processPrompts(options));
+  }.bind(this)).then(function (prompts) {
```

Comment:
An alternative, which Im totally on-board with is adding. `Promise.invoke` to our promise subclass

```
Promise.invoke(this._processPrompts.bind(this), options).then
```

which would likely be implemented as:

```js
function invoke(method, ...args) {
return new this(function(resolve) {
  resolve(method(...args));
});
```